From: Jan Kara <jack@suse.cz>
Subject: fs: Make extension of struct super_block transparent
Patch-mainline: Never, kabi
References: bsc#1117822

struct super_block is allocated through alloc_super() which gets proper struct
size. So adding entry at the end is fine.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/fs.h |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -1383,9 +1383,6 @@ struct super_block {
 	/* Number of inodes with nlink == 0 but still referenced */
 	atomic_long_t s_remove_count;
 
-	/* Pending fsnotify inode refs */
-	atomic_long_t s_fsnotify_inode_refs;
-
 	/* Being remounted read-only */
 	int s_readonly_remount;
 
@@ -1422,6 +1419,11 @@ struct super_block {
 
 	spinlock_t		s_inode_wblist_lock;
 	struct list_head	s_inodes_wb;	/* writeback inodes */
+
+#ifndef __GENKSYMS__
+	/* Pending fsnotify inode refs */
+	atomic_long_t s_fsnotify_inode_refs;
+#endif
 };
 
 /* Helper functions so that in most cases filesystems will
